src/openrct2/windows/ride_list.c

```diff
@@ -411,7 +421,13 @@ static void window_ride_list_scrollmousedown(rct_window *w, int scrollIndex, int
 		return;
 
 	// Open ride window
-	window_ride_main_open(w->list_item_positions[index]);
+	if (_quick_demolish_mode) {
+		gGameCommandErrorTitle = STR_CANT_DEMOLISH_RIDE;
+		game_do_command(0, 1, 0, w->list_item_positions[index], GAME_COMMAND_DEMOLISH_RIDE, 0, 0);
```

Comment:
I've addressed this in a later commit. I originally copied the code from the demolish ride prompt